docs/developer/architecture.rst

```diff
@@ -168,23 +168,41 @@ The base storage format in Modin is pandas. In the default case, the Modin Dataf
 Data Ingress
 ''''''''''''
 
+.. note::
+   Data ingress operations (e.g. ``read_csv``) in Modin load data from the source into
+   partitions and vice versa for data egress (e.g. ``to_csv``) operation.
+   Improved performance is achieved by reading/writing in partitions in parallel.
+
+The data ingress operations workflow diagram is shown below. After a data 
+ingress function from pandas API (e.g. ``read_csv``) is called, the compiled
+query from pandas API is passed to the :doc:`Factory Dispatcher </flow/modin/core/execution/dispatching>`,
+which provides IO class based on the selected execution. Resulting IO class will contain
```

Comment:
FactoryDispatcher defines a factory specific for the execution. The factory, in turn, exposes the IO class whose responsibility is to perform a parallel read/write from/to a file. Look at pandas on Ray section as an example and correct this text.

Reply:
Corrected.